test/test_images/probe_helper/probe_helper.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2020 Google LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"knative.dev/pkg/signals"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/protocol"
+	"github.com/kelseyhightower/envconfig"
+)
+
+type cloudEventsFunc func(cloudevents.Event) protocol.Result
+
+type envConfig struct {
+	// Environment variable containing the sink URL (broker URL) that the event will be forwarded to by the probeHelper
+	BrokerURL string `envconfig:"K_SINK" default:"http://default-brokercell-ingress.cloud-run-events.svc.cluster.local/cloud-run-events-probe/default"`
+
+	// Environment variable containing the port which listens to the probe to deliver the event
+	ProbePort int `envconfig:"PROBE_PORT" default:"8070"`
+
+	// Environment variable containing the port to receive the event from the trigger
+	ReceiverPort int `envconfig:"RECEIVER_PORT" default:"8080"`
+
+	// Environment variable containing the timeout period to wait for an event to be delivered back (in minutes)
+	Timeout int `envconfig:"TIMEOUT_MINS" default:"30"`
+}
+
+func forwardFromProbe(ctx context.Context, c cloudevents.Client, receivedEvents map[string] chan bool, timeout int) cloudEventsFunc{
+	return func(event cloudevents.Event) protocol.Result{
+		eventID := event.ID()
+		log.Print("Sending the cloud event to the broker")
+		receivedEvents[eventID] = make(chan bool, 1)
+		ctx, _ = context.WithTimeout(ctx, time.Duration(timeout) * time.Minute)
+		if res := c.Send(ctx, event); !cloudevents.IsACK(res) {
+			return cloudevents.ResultNACK
+		}
+		select {
+		case <-receivedEvents[eventID]:
+			delete(receivedEvents, eventID)
+			return cloudevents.ResultACK
+		case <-ctx.Done():
+			return cloudevents.ResultNACK
+		}
+	}
+}
+
+func receiveFromTrigger(receivedEvents map[string]chan bool) cloudEventsFunc {
+	return func(event cloudevents.Event) protocol.Result {
+		eventID := event.ID()
+		receivedEvents[eventID] <- true
```

Comment:
While testing, I found a small issue here. We should check if eventID exists in the map, if not, we just return ACK.

Let's do this in a follow up PR.

Reply:
I will create an issue regarding this.